calibrators/mapping_based_calibrator/README.md

```diff
@@ -0,0 +1,238 @@
+
+After the mapping has been completed, some preprocessing is necessary before estimating the transformations between lidars. In the explanation below, we assume that there is only a `calibration lidar` that needs to be calibrated with the `mapping lidar`.
+
+##### Data synchronization
+
+As the tool aims to apply registration algorithms on the pointclouds from both the `mapping lidar` and the `calibration lidar` to estimate the transformation between them, it's crucial to ensure those pointclouds share the same timestamp. However, the `calibration lidar` may not be synchronized with the `mapping lidar`, meaning these pointclouds cannot be directly used together during movement. To address this, the tool first needs to interpolate the pose of the `mapping lidar` at the timestamp of the `calibration lidar` in the map.
+
+##### Data selection
+
+To refine our calibration process, the tool first temporarily pairs `mapping lidar` keyframes with their closest `calibration lidar` frame, and then selects the final calibration pairs (`mapping lidar` keyframes and `calibration lidar` frame) based on the criteria listed below.
+
+- The pairs have low time difference and low interpolation error (such as speed, estimated acceleration, etc.) between the keyframe and the `calibration lidar` frame.
+- The pointlcoud in `calibration lidar` frame has enough features for calibration, which the tool filters out the pointcloud that has low variance in the z-axis that is mostly a plane.
+
+##### Data preprocessing
+
+For the `calibration lidar` pointcloud to have more points to apply the registration algorithm, the calibrator would need a high-resolution pointcloud from the `mapping lidar`. Therefore, we augment the keyframe's pointcloud with all the frames' pointcloud near it. As the resulting pointcloud has redundant information, we downsample the pointcloud to quite a fine-grained resolution, which is called augmented pointcloud.
+
+#### Step 3: Estimate transformations
+
+Once the data is prepared for calibration, the tool applies registration algorithms, such as NDT and GICP, to estimate the transformation between the pointcloud from the `calibration lidar` and the augmented pointcloud from the `mapping lidar`.
+
+### base-lidar calibration
+
+Instead of calibrating the transformations between lidars, we can also utilize the map generated by the `mapping lidar` to calibrate the transformation between the `mapping lidar` and the `base_link`.
+
+#### Step 1: Construct a map
+
+The first step of base-lidar calibration would be the same as [Step 1](#step-1-construct-a-map) in `lidar-lidar calibration`.
+
+#### Step 2: Extract ground plane from the pointcloud
+
+After constructing the map, we could get the augmented pointcloud from `mapping lidar`, which would be the same process as [Step 2](#step-2-prepare-calibration-data). Afterward, the tool utilizes PCA and a segmentation algorithm for extracting the ground plane from the augmented pointcloud.
```

Comment:
could?

Reply:
would?

Reply:
fixed in  https://github.com/tier4/CalibrationTools/pull/164/commits/3c261172913cc1608a57de0e985b26df28bc3005

Reply:
the step 2 in lidar-lidar calibration